vulnerability/events_creator.go

```diff
@@ -0,0 +1,167 @@
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package vulnerability
+
+import (
+	"strings"
+	"time"
+
+	db_types "github.com/aquasecurity/trivy-db/pkg/types"
+	trivy_types "github.com/aquasecurity/trivy/pkg/types"
+	"github.com/elastic/beats/v7/libbeat/beat"
+	libevents "github.com/elastic/beats/v7/libbeat/beat/events"
+	"github.com/elastic/cloudbeat/resources/providers/awslib/ec2"
+	"github.com/elastic/cloudbeat/transformer"
+	"github.com/elastic/cloudbeat/version"
+	"github.com/elastic/elastic-agent-libs/mapstr"
+)
+
+type Finding struct {
+	Target string                  `json:"target,omitempty"`
+	Type   string                  `json:"type,omitempty"`
+	Class  trivy_types.ResultClass `json:"class,omitempty"`
+	Vul    Vulnerability           `json:"vulnerability,omitempty"`
+}
+
+type Vulnerability struct {
+	Cvss           db_types.VendorCVSS  `json:"cvss,omitempty"`
+	DataSource     *db_types.DataSource `json:"data_source,omitempty"`
+	Scanner        Scanner              `json:"scanner,omitempty"`
+	Score          Score                `json:"score,omitempty"`
+	Package        Package              `json:"package,omitempty"`
+	Cwe            []string             `json:"cwe,omitempty"`
+	ID             string               `json:"id,omitempty"`
+	Title          string               `json:"title,omitempty"`
+	Enumeration    string               `json:"enumeration,omitempty"`
+	Reference      string               `json:"reference,omitempty"`
+	Description    string               `json:"description,omitempty"`
+	Severity       string               `json:"severity,omitempty"`
+	Classification string               `json:"classification,omitempty"`
+	PublishedDate  *time.Time           `json:"published_date,omitempty"`
+}
+
+type Package struct {
+	FixedVersion string `json:"fixed_version,omitempty"`
+	Version      string `json:"version,omitempty"`
+	Name         string `json:"name,omitempty"`
+}
+
+type Scanner struct {
+	Version string `json:"version,omitempty"`
+	Vendor  string `json:"vendor,omitempty"`
+}
+
+type Score struct {
+	Impact  float64 `json:"impact,omitempty"`
+	Base    float64 `json:"base,omitempty"`
+	Version string  `json:"version,omitempty"`
+}
+
+type Resource struct {
+	ID   string
+	Name string
+}
+
+const (
+	vulScannerVersion     = "v0.35.0"
+	vulScannerVendor      = "Trivy"
+	vulScoreSystemClass   = "CVSS"
+	vulScoreSource        = "nvd"
+	vulScoreSourceVersion = "3.0"
+	vulEcsCategory        = "vulnerability"
+	vulIndex              = "logs-cloud_security_posture.vulnerabilities-default"
+)
+
+func createVulnerabilityEvent(reportResult trivy_types.Result, vul trivy_types.DetectedVulnerability, ins ec2.Ec2Instance) beat.Event {
```

Comment:
Did you implement this based on the [technical index](https://github.com/elastic/security-team/blob/main/docs/cloud-security-posture-team/technical-index.mdx#vulnerability-finding-schema)?
I didn't find a field named `finding` on the index, I wonder if it's on purpose or not.